app/javascript/hyacinth_v1/components/users/UserEdit.js

```diff
@@ -44,6 +44,14 @@ function UserEdit() {
     return activatable;
   };
 
+  const canEditEmail = () => {
+    let editEmail = ability.can('manage', 'all');
+    if (!editEmail) {
+      editEmail = (ability.can('manage', 'User') && !isAdmin);
+    }
+    return editEmail;
+  };
```

Comment:
It would be good to double check me on this suggestion, but would this work?

```suggestion
  const canEditEmail = () => {
    return ability.can('manage', 'all') || ability.can('manage', 'User');
  };
```

If you can `manage all` then you're an admin, otherwise only allow if you can `manage User` (which means you're a user admin).  Otherwise this would result in false.

Reply:
Confusingly, the !isAdmin value refers to the user record being edited rather than the user who is editing. So this line does what we want:

  const canEditEmail = () => ability.can('manage', 'all') || (ability.can('manage', 'User')  && !isAdmin);

Would it make sense to change the name of that variable, maybe to !isAdminRecord?

Reply:
Oh, right!  I just took a look at the whole file and I see what you mean.  I'm okay with keeping it as just `isAdmin` because of the way the other top variables refer to the current record rather than the ability of the viewer.  And I like your suggested one-line change if you're okay with that instead of the `if` statement.  (For me, it's slightly faster to read because it's more compact.)

Reply:
I'll approve this PR now and leave the line change up to you.  I'm good either way.  Thanks for the PR!